hud.lua

```diff
@@ -731,66 +708,62 @@ function courseplay.hud:updatePageContent(vehicle, page)
 					else
 						self:disableButtonWithFunction(vehicle,page, 'cancelWait')
 					end				
-				elseif entry.functionToCall == 'changeStartAtPoint' then
+				elseif entry.functionToCall == 'Setting:startingPoint:next' then
+					--StartingPointSetting
 					if not vehicle:getIsCourseplayDriving() and vehicle.cp.canDrive then
-						self:enableButtonWithFunction(vehicle,page, 'changeStartAtPoint')
+						self:enableButtonWithFunction(vehicle,page, 'Setting:startingPoint:next')
```

Comment:
Instead of this and that change in executeFunction, could we just pass in the object and the function call? So all we have to do is just to do object[function](object, ...)

Reply:
sound like good idea, but what should we do about this: button => setCourseplayFunc() => CourseplayEvent.sendEvent() => executeFunction() => object[function](object, ...) ..
might be tricky to pass the settings object throw multiplayer sync as it is right now

Reply:
Honestly, I don't really see what exactly this event is (and for that matter, what is the intention of the setCourseplayFunc() function is. There may be a better way to do what these are doing. Can you explain what are these doing?

Reply:
It's related to the current buttons as they only pass one string as "funtion name" and one parameter to setCourseplayFunc from their it will be decided if all the client have to run the same function Call and after that is "executeFunction()" called to put the functionString back to a funtionCall. 
The problem would be to figure out a way to send the setting object or a synonym for it , throw to the other clients.